docs: edit attribute-binding doc and move colSpan note to property binding (#38860)
angular/angular: aio/content/guide/attribute-binding.md

# Attribute, class, and style bindings

Attribute binding in Angular helps you set values for attributes directly.
With attribute binding, you can do things such as improve accessibility, style your application dynamically, and manage multiple CSS classes or styles simultaneously.

<div class="alert is-helpful">

See the <live-example></live-example> for a working example containing the code snippets in this guide.

</div>

## Binding to an attribute

Usually, setting an element property with a [property binding](guide/property-binding) is preferable to setting the attribute with a string.
However, sometimes there is no element property to bind, so attribute binding is the solution.

For example, [ARIA](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA) and
[SVG](https://developer.mozilla.org/en-US/docs/Web/SVG) are purely attributes.
Neither ARIA nor SVG correspond to element properties and don't set element properties.
In these cases, you must use attribute binding because there are no corresponding property targets.

## Syntax

> Attribute binding syntax resembles [property binding](guide/property-binding), but instead of an element property between brackets, start with the prefix `attr`, followed by a dot, and the name of the attribute.

Reviewer: ```suggestion
Attribute binding syntax resembles [property binding](guide/property-binding), but instead of an element property between brackets, you precede the name of the attribute with the prefix `attr`, followed by a dot.
```

> Then set the attribute value with an expression that resolves to a string.

Reviewer: ```suggestion
Then, you set the attribute value with an expression that resolves to a string.
```

For example, [ARIA](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA) and
[SVG](https://developer.mozilla.org/en-US/docs/Web/SVG) are purely attributes.
Neither ARIA nor SVG correspond to element properties and don't set element properties.
In these cases, you must use attribute binding because there are no corresponding property targets.

## Syntax

Attribute binding syntax resembles [property binding](guide/property-binding), but instead of an element property between brackets, start with the prefix `attr`, followed by a dot, and the name of the attribute.
Then set the attribute value with an expression that resolves to a string.

<code-example language="html">

 &lt;p [attr.attribute-you-are-targeting]="expression"&gt;&lt;/p&gt;

</code-example>

<div class="alert is-helpful">

When the expression resolves to `null`, Angular removes the attribute altogether.

</div>

## Seting ARIA attributes

One of the primary use cases for attribute binding
is to set ARIA attributes, as in this example:

<code-example path="attribute-binding/src/app/app.component.html" region="attrib-binding-aria" header="src/app/app.component.html"></code-example>

{@a colspan}

> ## Using attribute binding with `colspan`

Reviewer 2: I think `colSpan` is just an example of a case where attribute and property names are different and there are more cases like that. For example the same applies to the [rowSpan property vs rowspan attribute](https://www.w3schools.com/jsref/prop_tabledata_rowspan.asp). May be we can have a high-level description of this situation here (that attribute and property names are not always the same) and provide a link to the `colSpan` example case that we haven at guide/property-binding#colspan? That should help make this page more simple and keep the `colSpan` example details in just one place.

Author: Agree 100%! I've moved it to `property-binding`, added a statement about the potential difference, and linked.

To use attribute binding with the `<td>` attribute `colspan`, specify the `colspan` attribute by prefixing it with `attr` followed by a dot, and placing `attr.colspan` within square brackets.

> Next, set `[attr.colspan]` equal to an expression.

Author: Yes, I believe it is to make the sizing of a table dynamic. I'll add it. :)

> Next, set `[attr.colspan]` equal to an expression.

Author: Edited to:

```
Another common use case for attribute binding is with the `colspan` attribute in tables.
Binding to the `colspan` attribute helps you keep your tables programmatically dynamic.
Depending on the amount of data that your application populates a table with, the number of columns that a row spans could change.
```

This causes the `<tr>` to span two columns.

<div class="alert is-helpful">

`colspan` is an attribute of `<tr>`, while `colSpan`  with a capital "S" is a property.
When using attribute binding, use `colspan` with a lowercase "s".
For more information on how to bind to the `colSpan` property, see the [`colspan` and `colSpan`](guide/property-binding#colspan) section of [Property Binding](guide/property-binding).

</div>

<hr/>

{@a class-binding}

## Binding to the `class` attribute

Use class binding to add and remove CSS class names from an element's `class` attribute.

### Binding to a single CSS `class`

To create a single class binding, use the prefix `class` followed by a dot and the name of the CSS class&mdash;for example, `[class.sale]="onSale"`.
Angular adds the class when the bound expression, `onSale` is truthy, and it removes the class when the expression is falsy&mdash;with the exception of `undefined`.
See [styling delegation](#styling-delegation) for more information.

### Binding to multiple CSS classes

To bind to multiple classes, use `[class]` set to an expression&mdash;for example, `[class]="classExpression"`.

> The expression&mdash;or what is between the quotation marks&mdash;can be a space-delimited string of class names, or an object with class names as the keys and truthy/falsy expressions as the values.

Author: Agreed. I removed the explanation of what the expression is because it's explained above. And I took away the slash.

{@a colspan}

## Binding to `colspan`

Another common use case for attribute binding is with the `colspan` attribute in tables.
Binding to the `colspan` attribute helps you keep your tables programmatically dynamic.
Depending on the amount of data that your application populates a table with, the number of columns that a row spans could change.

To use attribute binding with the `<td>` attribute `colspan`:

> 1. Specify the `colspan` attribute by prefixing it with `attr` followed by a dot, and placing `attr.colspan` within square brackets.

Reviewer: ```suggestion
1. Specify the `colspan` attribute by using the following syntax: `[attr.colspan]`.
```

1. Set `[attr.colspan]` equal to an expression.

> In the following example the target is `[attr.colspan]` and it is equal to the expression `1 + 1`.

Reviewer: ```suggestion
In the following example, binds the `colspan` attribute to the expression `1 + 1`.
```

<code-example path="attribute-binding/src/app/app.component.html" region="colspan" header="src/app/app.component.html"></code-example>

This binding causes the `<tr>` to span two columns.

<div class="alert is-helpful">

Sometimes there are differences between the name of property and an attribute.

`colspan` is an attribute of `<tr>`, while `colSpan`  with a capital "S" is a property.
When using attribute binding, use `colspan` with a lowercase "s".
For more information on how to bind to the `colSpan` property, see the [`colspan` and `colSpan`](guide/property-binding#colspan) section of [Property Binding](guide/property-binding).

</div>

<hr/>

{@a class-binding}

## Binding to the `class` attribute

> Use class binding to add and remove CSS class names from an element's `class` attribute.

Reviewer: ```suggestion
You can use class binding to add and remove CSS class names from an element's `class` attribute.
```

  </tr>
</table>

<hr/>

{@a style-binding}

> ## Binding to the style attribute

Reviewer 2: It feels like the whole styling section deserves its own page :) While it's technically related to the attribute bindings, there is a lot of styling-specific information (which is ~80% of the page content) which makes this page heavy to read in case I just want to get more info on attribute bindings.

This is probably outside of the scope of this PR, but I'd propose moving styling-related docs to a separate page to keep the attribute binding page simple.

Author: Very insightful! I've added it to our notes on this PR. Thanks, @AndrewKushnir!
cc @aikidave

Reviewer 2: Thanks @kapunahelewong. I'd also propose merging attribute and property binding pages, it looks like we refer from one page to another, so having this information on a single page might be beneficial. So we might have the following sections under the "Main Concepts" group:
- Attribute and Property bindings (which might contain a reference to the Styling page below when it comes to `[class]` and `[style]` bindings)
- Styling

Reviewer: These are good suggestions! @kapunahelewong, could you create two issues: one for moving the styling content, and another for merging the attribute and property binding pages? We can then look into each more fully!

Author: https://github.com/angular/angular/issues/39141 and https://github.com/angular/angular/issues/39160

Use style binding to set styles dynamically.

### Binding to a single style

To create a single style binding, use the prefix `style` followed by a dot and the name of the CSS style property&mdash;for example, `[style.width]="width"`.
Angular sets the property to the value of the bound expression, which is usually a string.
Optionally, you can add a unit extension like `em` or `%`, which requires a number type.

<div class="alert is-helpful">

You can write a style property name in either [dash-case](guide/glossary#dash-case), or
[camelCase](guide/glossary#camelcase).

</div>

### Binding to multiple styles

> To toggle multiple styles , bind to the `[style]` attribute&mdash;for example, `[style]="styleExpression"`).

Reviewer 2: ```suggestion
To toggle multiple styles, bind to the `[style]` attribute&mdash;for example, `[style]="styleExpression"`).
```

  </tr>
</table>

{@a styling-precedence}

> ## Style Precedence

Reviewer: This section seems fine, but again, it's very conceptual. Consider creating a new topic and moving it to the Conceptual reference?
